src/components/MoneyRequestHeader.js

```diff
@@ -125,5 +132,15 @@ export default compose(
         parentReport: {
             key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`,
         },
+        parentReportActions: {
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.parentReportID}`,
+            canEvict: false,
+        },
+    }),
+    withOnyx({
+        transaction: {
+            key: ({report, parentReportActions}) =>
+                `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(parentReportActions, [report.parentReportActionID, 'originalMessage', 'IOUTransactionID'], '')}`,
+        },
     }),
```

Comment:
It's composing, since we need parentReportActions (calculated before). This came from this discussion: https://github.com/Expensify/App/pull/25498#discussion_r1303616459

Reply:
I followed a little bit in the previous discussion, but I think this is wrong. We should not be using duplicate `withOnyx` and I haven't found a case where it is necessary yet. 

[I'm doing a small refactoring](https://github.com/Expensify/App/pull/27286/files) to `MoneyRequestHeader.js` and it works fine, plus, `parentReportActions` is not needed because the proper `parentReportAction` is already attained in one of the other props being loaded from Onyx.

[Let's discuss this further in Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1694710295343199). (@BeeMargarida I could not find you on Slack, but can you please join the convo?)